api/profile/favoritesRouter.js

```diff
@@ -0,0 +1,47 @@
+const express = require('express');
+const authRequired = require('../middleware/authRequired');
+const router = express.Router();
+const Model = require('../globalModel');
+
+router.get('/:id/favorites', authRequired, function (req, res) {
+  const id = String(req.params.id);
+  Model.favorites('favorite_category', id)
```

Comment:
looks like there's a signature mismatch between here and the function definition in global model, that function only accepts an id param, not a table name. Is it working correctly like this? seems likely to cause the query to always return no results.